pennylane/utils.py

```diff
@@ -385,3 +385,44 @@ def circuit2():
         qml.QueuingContext.append_operator(inv_op)
 
     return inv_ops
+
+
+def expand_matrix(matrix, original_wires, expanded_wires):
+    r"""Expand a an operator matrix to more wires.
+
+    Args:
+        matrix (array): :math:`2^n \times 2^n` matrix where n = len(original_wires).
+        original_wires (Sequence[int]): original wires of matrix
+        expanded_wires (Sequence[int]): expanded wires of matrix, can be shuffled
+
+    Returns:
+        array: :math:`2^m \times 2^m` matrix where m = len(expanded_wires).
+    """
+    N = len(original_wires)
+    M = len(expanded_wires)
+    D = M - N
+
+    dims = [2] * (2 * N)
+    tensor = matrix.reshape(dims)
+
+    if D > 0:
+        extra_dims = [2] * (2 * D)
+        identity = np.eye(2 ** D).reshape(extra_dims)
+        expanded_tensor = np.tensordot(tensor, identity, axes=0)
+        # Fix order of tensor factors
+        expanded_tensor = np.moveaxis(expanded_tensor, range(2 * N, 2 * N + D), range(N, N + D))
+    else:
+        expanded_tensor = tensor
+
+    wire_indices = []
+    for wire in original_wires:
+        wire_indices.append(expanded_wires.index(wire))
+
+    wire_indices = np.array(wire_indices)
+
+    # Order tensor factors according to wires
+    original_indices = np.array(range(N))
+    expanded_tensor = np.moveaxis(expanded_tensor, original_indices, wire_indices)
+    expanded_tensor = np.moveaxis(expanded_tensor, original_indices + M, wire_indices + M)
+
+    return expanded_tensor.reshape((2 ** M, 2 ** M))
```

Comment:
Nice! Is the previous `expand` function being used anywhere? If not, might make sense to replace it now (I recall it used to be used by default.qubit, before default.qubit was rewritten to use tensordot)

Reply:
It's also used in `QubitQNode` ;)

Reply:
I now changed the signature of the new expand function to also support the call structure of the old one 👍